setup.py

```diff
@@ -2,8 +2,89 @@
 # -*- encoding: utf-8 -*-
 
 from setuptools import setup
+from distutils.command import build
+from distutils.command import install
+#from distutils.command import install_scripts
+import distutils
+import subprocess
+
 from kiwi.version import __version__
 
+
+class kiwi_build(build.build):
+    """
+    Custom build command
+    """
+    build.build.user_options += [
+        ('cflags=', None, 'compile flags')
+    ]
+
+    def initialize_options(self):
+        """
+        Set default values for options
+        Each user option must be listed here with their default value.
+        """
+        build.build.initialize_options(self)
+        self.cflags = ''
+
+    def run(self):
+        """
+        Run first the related KIWI C compilation and after that
+        the usual Python build
+        """
+        # kiwi C tools compilation
+        command = ['make']
+        if self.cflags:
+            command.append('CFLAGS=%s' % self.cflags)
+        command.append('tools')
+        self.announce(
+            'Running make tools target: %s' % str(command),
+            level=distutils.log.INFO
+        )
+        self.announce(
+            subprocess.check_output(command).decode(),
+            level=distutils.log.INFO
+        )
+
+        # standard build process
+        build.build.run(self)
+
+
+class kiwi_install(install.install):
+    """
+    Custom install command
+    """
+    sub_commands = [
+        ('install_lib', lambda self:True),
+        ('install_headers', lambda self:False),
+        ('install_scripts', lambda self:True),
+        ('install_data', lambda self:False),
+        ('install_egg_info', lambda self:True),
+    ]
+
+    def run(self):
```

Comment:
Ok, I've started from scratch. Run "./setup.py build" and after that "./setup.py install --prefix /tmp/kiwi". This gives me a slightly different version (note the absolute paths):

```
$ ./setup.py install --prefix /tmp/kiwi
running install
Running make install target: ['make', 'install']
install: cannot create regular file ‘/usr/bin/kversion’: Permission denied
install: cannot create regular file ‘/usr/bin/utimer’: Permission denied
install: cannot create regular file ‘/usr/bin/dcounter’: Permission denied
install: cannot create regular file ‘/usr/bin/startshell’: Permission denied
install: cannot create regular file ‘/usr/bin/isconsole’: Permission denied
install: cannot create regular file ‘/usr/bin/driveready’: Permission denied
install: cannot create regular file ‘/usr/bin/setctsid’: Permission denied
make[1]: *** [install] Error 1
```

I'm not sure what caused this. I will try it on a different machine and try it there.